core/src/main/java/io/questdb/griffin/engine/functions/groupby/StringAggGroupByFunction.java

```diff
@@ -181,9 +225,36 @@ public void setNull(MapValue mapValue) {
         mapValue.putBool(valueIndex + 1, true);
     }
 
+    @Override
+    public void setEmpty(MapValue mapValue) {
+        mapValue.putLong(valueIndex, 0);
+        mapValue.putBool(valueIndex + 1, true);
+    }
+
     @Override
     public boolean supportsParallelism() {
-        return false;
+        return true;
+    }
+
+    @Override
+    public boolean isThreadSafe() {
+        return UnaryFunction.super.isThreadSafe();
```

Comment:
This function can't be marked as thread-safe as it uses both flyweight sinks and thread-unsafe group by allocator.